networkx/algorithms/community/nodal_roles_nx.py

```diff
@@ -0,0 +1,128 @@
+from networkx.algorithms.community import louvain
+import networkx as nx
+
+def convert_dictionary(dictionary):
+	'''
+	Converts dictionary with nodes as keys to modules as keys
+
+	------
+	Inputs
+	------
+	dictionary =  dictionary output from Louvain community detection
+
+	------
+	Output
+	------
+	Dictionary with modules as keys, nodes as values
+
+	'''
+
+	new_dict = {}
+	for m,n in zip(dictionary.values(),dictionary.keys()):
+	    try:
+	        new_dict[m].append(n)
+	    except KeyError:
+	        new_dict[m] = [n]
+	return new_dict
+
+def get_within_module_edges(module, graph):
+	module_edges = []
+	for edge in graph.edges():
+		if edge[0] in module and edge[1] in module:
+			module_edges.append(edge)
+	return module_edges
+
+
+def within_module_degree(graph, partition, weighted = False):
+    '''
+    Computes the within-module degree for each node (Guimera et al. 2005)
+
+    ------
+    Inputs
+    ------
+    graph = Networkx Graph
+   	partition = dictionary from modularity partition of graph using Louvain method
+    where the keys are nodes and values are module numbers
+    weighted: Boolean
+
+    ------
+    Output
+    ------
+    Dictionary of the within-module degree of each node.
+
+    '''
+    graph = graph.to_undirected()
+    partition = convert_dictionary(partition)
+    wmd_dict = {}
+    for module in partition.keys():
+        module = partition[module]
+        module_wd_dict = {}
+        module_edges = get_within_module_edges(module,graph)
+        for source in module:
+            edge_count = 0
+            for target in module:
```

Comment:
Maybe simpler, something like: `edge_count = sum(graph.edge[s][t]['weight'] or 1 for target in module if (s, t) in module_edges or (t, s) in module_edges)`.